### DISCUSSION TITLE
Is it appropriate to create a connection for every query?The number of queries per day is about 50000

### DISCUSSION THREAD

**Discussion by zyy20191** [milvus-io/milvus#18116] [Q&A and General discussion] [answered]
After testing, I found that each new connection query speed is the fastest.

Our service has about 50,000 queries per day. 
If each query creates a new connection, will Milvus go down or something? 
Is there a recommended connection method for a large number of queries?
Is it feasible to create a new connection for each query?

**Comment by yhmo** [comment, marked as answer]
No need to create connection for each query. It will waste your system's resource if you created too many connections, you might hit the ulimit of linux os.
One connection for one thread is recommended.
